src/huggingface_hub/hf_file_system.py

```diff
@@ -649,6 +654,58 @@ def _upload_chunk(self, final: bool = False) -> None:
             )
 
 
+class HfFileSystemStreamFile(fsspec.spec.AbstractBufferedFile):
+    def __init__(
+        self, fs: HfFileSystem, path: str, revision: Optional[str] = None, block_size=0, cache_type="none", **kwargs
+    ):
+        if block_size != 0:
+            raise ValueError(f"HfFileSystemStreamFile only supports block_size=0 but got {block_size}")
```

Comment:
```suggestion
            raise ValueError(f"HfFileSystemStreamFile only supports block_size=0 but got '{block_size}'.")
```

Reply:
Is it a restriction from us or from `fsspec`? (if from us, let's raise `NotImplementedError`)

Reply:
`HfFileSystemStreamFile` should never be called with these arguments (`open()` chooses the right file class to instantiate based on those), so it's a ValueError